models/webhook.go

```diff
@@ -256,6 +259,21 @@ func (w *Webhook) EventsArray() []string {
 	return events
 }
 
+func (w *Webhook) checkBranch(branch string) bool {
+	if w.BranchFilter == "" {
```

Comment:
Might be good to optimize `*` glob to have less calls
```suggestion
	if w.BranchFilter == "" || w.BranchFilter == "*" {
```

Reply:
It depends whether `*` matches all files (deep match) or only the root directory (shallow match). This library (gobwas) permits both scenarios. If you make the `/` character special (i.e., it is a path delimiter), to match all files the glob pattern is `**` and not `*`; this is the default, although it can be changed.

Should we specify `/` as a special character? I'd say yes; otherwise `my*doc` will match `mytext.doc` as well as `myfolder/iscool.doc`, which might be confusing.

Check [my comment in this PR](https://github.com/go-gitea/gitea/pull/7767#issuecomment-518946214).

Reply:
I don't think we need to treat `/` as special character as we are not dealing with paths here. Also as `*` is default it should match all branches even with `/` in them otherwise it will be breaking feature

Reply:
Sorry, you are right!

Reply:
Done. I also updated tests to also ensure that `feature*` matches `feature/7791`.